src/lwc/geFormWidgetAllocation/geFormWidgetAllocation.js

```diff
@@ -153,79 +161,183 @@ export default class GeFormWidgetAllocation extends LightningElement {
-                    let diSourceField = dataImportRow[diKey].sourceObj[fieldMappings[fieldMappingKey].Source_Field_API_Name];
-                    
-                    properties[targetField] = diSourceField;
+                if (Object.keys(sourceObj).includes(sourceField)) {
+                    targetField = [fieldMappings[fieldMappingKey].Target_Field_API_Name];
+                    diSourceField =
+                        dataImportRow[diKey].sourceObj[fieldMappings[fieldMappingKey].Source_Field_API_Name];
+
+                    record[targetField] = diSourceField;
 
                 }
             });
-            rowList.push(properties);
+            targetRecords.push(record);
         });
-        
-        this.addRows(rowList);
+        this.addRows(targetRecords);
+    }
+
+    /**
+     * @description This method determines if the data object loaded into the widget was loaded from a data import
+     *              row.
+     * @returns Boolean
+     */
+    hasDonationImported(data) {
+        return data.hasOwnProperty(DATA_IMPORT_DONATION_IMPORTED_FIELD.fieldApiName) &&
+               isNotEmpty(data[DATA_IMPORT_DONATION_IMPORTED_FIELD.fieldApiName]);
+    }
+
+    /**
+     * @description This method determines if the data object loaded into the widget is in the target object format.
+     * @returns Boolean
+     */
+    hasAdditionalObjectJSON(data) {
+        return Object.keys(data).includes(DI_ADDITIONAL_OBJECT.fieldApiName);
     }
 
     /**
      * Handle Add Row being clicked
      */
     handleAddRow() {
-        this.addRow(false);
+        this.addRows(
+            [this.getRow(false)]
+        );
     }
 
+    /**
+     * Add a new record to the list
+     * @description Adds a collection of widget row objects to the row list in order to re-render the widget and
+     * display the new rows
+     * @param isDefaultGAU {boolean} When initializing the first row, this should be true.
+     */
     addRows(records) {
+        let rows = [];
         records.forEach(record => {
-            this.addRow(false, record);      
+            if(record.isDefaultGAU) {
+                rows.push(record);
+            } else {
+                rows.push(this.getRow(false, record));
+            }
         });
+        this.rowList = this.rowList.concat(rows);
+        // Force the default GAU to reallocate to accommodate for unpredictable rendering behaviors
+        this.allocateDefaultGAU();
     }
 
     /**
-     * Add a new record to the list
+     * @description Build a new row object using both record and element properties
      * @param isDefaultGAU {boolean} When initializing the first row, this should be true.
+     * @param properties {array} Array of record properties to add to the record object
+     * @return Object representing a widget row
      */
-    addRow(isDefaultGAU, properties) {
+    getRow(isDefaultGAU, properties) {
```

Comment:
`getRow` implies a row already exists to me, how about `buildRow` or `generateRow`?

Reply:
fixed